### PR TITLE
[VL-144] - TC-Send02 checks

### PR DESCRIPTION
**Pull request by kin0992**
Add some checks required in order to satisfy the test case tc-send02.

Added checks:
- verify a stream has been created
- verify that the endpoint to get the events of a stream has been called using the streamId provided by the first endpoint
- verify the accepted event has been received
- verify that an event with a iun populated has been received

Missing checks (that we are going to implements in a future work):
- verify the events successive of the accepted have been received
- verify the endpoints have been called honouring the retry-after

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Documentation has been updated in #85

### CODE REVIEW

**Comment by kin0992** [comment]
At the moment, this TC verifies only the conditions that there is an event with the newStatus set to "ACCEPTED" and there is an event with the IUN set.

We are waiting for more information requested [here](https://pagopa.atlassian.net/wiki/spaces/PN/pages/560595566/Test+di+certificazione+avvenuta+integrazione+con+Piattaforma+Notifiche?focusedCommentId=583991603) and then we can implement the last mile and convert this PR to ready for review. 
cc @datalek

File: src/domain/CreateEventStreamRecordRepository.ts [hunk 1]
```diff
@@ -11,3 +14,20 @@ export type CreateEventStreamRecord = AuditRecord & {
...
 
 export type CreateEventStreamRecordRepository = Repository<CreateEventStreamRecord>;
+
+export const isCreateEventStreamRecord = (record: AllRecord): O.Option<CreateEventStreamRecord> =>
+  record.type === 'CreateEventStreamRecord' ? O.some(record) : O.none;
+
+export const hasTimelineEventTypeToTimeline = (record: CreateEventStreamRecord) =>
+  record.input.body.eventType === EventTypeEnum.TIMELINE;
+
+export const isSuccessfulResponse = (record: CreateEventStreamRecord) => record.output.statusCode === 200;
```
Comment (datalek) [inline src/domain/CreateEventStreamRecordRepository.ts:24]: 🤔 I'm not sure if the method `isSuccessfulResponse` is worth, it is used only in one place and somehow tested multiple times. Am I a bad human being (😈) if I think we should remove it? WDYT?

File: src/domain/CreateEventStreamRecordRepository.ts [hunk 2]
```diff
@@ -11,3 +14,20 @@ export type CreateEventStreamRecord = AuditRecord & {
...
+
+export const existsCreateEventStreamRecordWhitStreamId =
```
Comment (datalek) [inline src/domain/CreateEventStreamRecordRepository.ts:26]: At some point, we should find a consistent place for these predicates. They are sometimes in the `*Repository` files, and sometimes in the `checks/*Checks` files :persevere:

File: src/domain/CreateEventStreamRecordRepository.ts (hunk 1 above)
Comment (kin0992) [reply src/domain/CreateEventStreamRecordRepository.ts:24]: Yeah, agree. Done!

File: src/domain/CreateEventStreamRecordRepository.ts (hunk 2 above)
Comment (kin0992) [reply src/domain/CreateEventStreamRecordRepository.ts:26]: I agree! We could track this in a future work

**Review by datalek — APPROVED** [review]
LGTM 👍 . I've left some comments, but we are going to fix them in the next sprint.

**Pull request merged**